tests/unit/test_results.py

```diff
@@ -0,0 +1,77 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from tmt.cli import TmtExitCode
+from tmt.result import ResultOutcome, results_to_exit_code
+
+
+@pytest.mark.parametrize(
+    ('outcomes', 'expected_exit_code'),
+    [
+        # "No test results found."
+        (
+            [],
+            TmtExitCode.NO_RESULTS_FOUND
+            ),
+        # "Errors occured during test execution."
+        (
+            [
+                ResultOutcome.PASS,
+                ResultOutcome.FAIL,
+                ResultOutcome.INFO,
+                ResultOutcome.SKIP,
+                ResultOutcome.WARN,
+                ResultOutcome.ERROR
+                ],
+            TmtExitCode.ERROR
+            ),
+        # "There was a fail or warn identified, but no error."
+        (
+            [
+                ResultOutcome.PASS,
+                ResultOutcome.FAIL,
+                ResultOutcome.INFO,
+                ResultOutcome.SKIP,
+                ResultOutcome.WARN,
+                ],
+            TmtExitCode.FAIL
+            ),
+        # "Test were executed, and all reported ``skip`` result."
+        (
+            [
+                ResultOutcome.SKIP,
+                ResultOutcome.SKIP,
+                ],
+            TmtExitCode.ALL_RESULTS_SKIPPED
+            ),
+        # "At least one test passed, there was no fail, warn or error."
+        (
+            [
+                ResultOutcome.PASS,
+                ResultOutcome.INFO,
+                ResultOutcome.SKIP,
+                ],
+            TmtExitCode.SUCCESS
+            ),
+        # The catch-the-rest exit code
+        (
+            [
+                ResultOutcome.INFO,
+                ResultOutcome.SKIP,
+                ],
+            TmtExitCode.ERROR
```

Comment:
Hm, according to the [spec](https://tmt.readthedocs.io/en/latest/spec/plans.html#spec-plans-results) `info` should be handled as `pass`. So, I guess, we should give `SUCCESS` here as well?

Reply:
Fixed in b0cbb07c

Reply:
Ok, this does indeed change the current behavior, I suppose `breakage` is in order:

```
:: [ 15:53:38 ] :: [  BEGIN   ] :: Result: info, Test name: /test/always-info, Original result: 'pass', tmt return code: 2 :: actually running 'tmt run -a --scratch --id ${run} test --name /test/always-info provision --how local report -v 2>&1 >/dev/null | grep report -A2 | tail -n 1'
            info /test/always-info (original result: pass)
:: [ 15:53:40 ] :: [   FAIL   ] :: Result: info, Test name: /test/always-info, Original result: 'pass', tmt return code: 2 (Expected 2, got 0)
```

Reply:
After a quick sync, the exit code changes. Release note added in 2d729c66.